src/logstash-filters/snippets/doppler.conf

```diff
@@ -0,0 +1,71 @@
+# Parse Cloud Foundry logs from doppler (via https://github.com/SpringerPE/firehose-to-syslog)
+
+json {
+    source => 'syslog_message'
+    add_tag => [ 'cloudfoundry_doppler' ]
+}
+
+if "cloudfoundry_doppler" in [tags] {
+
+    date {
+        match => [ "time", "ISO8601" ]
+    }
+
+    if ('RTR' in [source_type]) {
+        grok {
+            match => {
+                'msg' => '%{HOSTNAME:hostname} - \[(?<time>%{MONTHDAY}/%{MONTHNUM}/%{YEAR}:%{TIME} %{INT})\] \"%{WORD:verb} %{URIPATHPARAM:path} %{PROG:http_spec}\" %{BASE10NUM:status} %{BASE10NUM:body_bytes_sent} \"%{GREEDYDATA:referer}\" \"%{GREEDYDATA:http_user_agent}\" %{HOSTPORT:delete_me} x_forwarded_for:\"%{GREEDYDATA:x_forwarded_for}\" vcap_request_id:%{NOTSPACE:vcap_request_id} response_time:%{NUMBER:response_time} app_id:%{NOTSPACE:delete_me}'
+                overwrite => [ "time" ]
+                'tag_on_failure' => [ 'fail/cloudfoundry/doppler/RTR' ]
+            }
+        }
+
+        if !("fail/cloudfoundry/doppler/RTR" in [tags]) {
+            date {
+                match => [ "time", "dd/MM/y:HH:mm:ss Z" ]
+            }
+            if [x_forwarded_for] {
+                mutate {
+                    gsub => ["x_forwarded_for","[\s\\"]",""] # remove quotes and whitespace
+                    split => ["x_forwarded_for", ","] # format is client, proxy1, proxy2 ...
+                }
+
+               mutate {
+                  add_field => ["remote_addr", "%{x_forwarded_for[0]}"]
+               }
+                            
+               geoip {
+                 source => "remote_addr"
+               }
+            }
+
+            mutate {
+                remove => [ "log_data", "delete_me", "msg" ]
```

Comment:
Rather than this remove option, don't name the fields in the grok pattern - e.g. instead of `%{HOSTPORT:delete_me}` use `%{HOSTPORT}`.

Actually, you can get rid of the whole section if you add the typing to the grok pattern - e.g. instead of `%{BASE10NUM:status}` use `%{BASE10NUM:status:int}`.